### ISSUE TITLE
Horizontal Rule / Divider Component

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-core#382]
# Issue Description
We want to provide a component to easily create a horizontal divider. 

## Issue Type

- [x] New Feature
- [ ] Enhancement
- [ ] Bug
- [ ] Other

## Expected Behavior
A react component which renders a horizontal rule. It should have props to change the color of the ruler and the height of the rule. By default, the height should be 1px.

**Comment by bjankord** [comment]
Here is some sample markup to render with the react component

```
<hr class="divider" />
```

and here is some sample css to go with it

```
.divider {
    border: 0 solid;
    border-top: 1px solid #dedfe0;
    margin: 0;
    height: 0;
}
```

**Comment by msaikaushik1992** [comment]
# Tech Design 
## Summary:
The horizontal rule/divider component is a component that is used to separate content on a layout.

## Responsiveness:
N/A

## Accessibility
N/A

## React Props:
N/A

## Theme-able styles:
borer-color
border-width

**Comment by bjankord** [comment]
+1 on tech design.

**Comment by mjhenkes** [comment]
+1

**Comment by JakeLaCombe** [comment]
+1. Is it possible to move this to our internal tech designs repo as well?

**Comment by msaikaushik1992** [comment]
After discussing in the war room about this, we all decide to use github issues itself for tech-design unless the issue is too complex @JakeLaCombe

**Comment by emilyrohrbough** [comment]
Resolved in #962
